src/components/GlobalPreferences/Notifications/NotificationsLogin.js

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import { Box, Button, GU, TextInput, Info, IconMail } from '@aragon/ui'
+import { login } from './notification-service-api'
+import { getEthNetworkType } from '../../../local-settings'
+import notificationSvg from './notifications.svg'
+
+export default function NotificationsLogin({
+  dao,
+  authState,
+  email,
+  onEmailChange,
+}) {
+  const [inputEmail, setInputEmail] = useState('')
+  const [apiError, setApiError] = useState(null)
+  const network = getEthNetworkType()
+
+  const handleEmailChange = e => {
+    setInputEmail(e.target.value)
+  }
+
+  const handleLogin = async e => {
+    try {
+      await login({ email: inputEmail, dao, network })
+      onEmailChange(inputEmail)
+    } catch (e) {
+      setApiError(e.message)
+      console.error('Failed to login', e)
+    }
+  }
+
+  return (
+    <Box heading="Email notifications">
+      <NotificationImage />
+      {apiError && <Info mode="error">Error logging in.</Info>}
+      <Label>Email address</Label>
+      <div
+        css={`
+          display: grid;
+          align-items: center;
+          grid-template-columns: 1fr auto;
+          grid-gap: ${2 * GU}px;
+          margin-bottom: ${2 * GU}px;
+        `}
```

Comment:
I recommended grid in general, and one of the reason why I choose `grid` over independent `margin`s on elements is for responsive behaviour: some times for smaller viewports we want different alignments or different spacing and with `grid` I find it easier to set the "layout" logic in one place instead of looking at what diff elements are doing.